src/libcharon/plugins/kernel_super/kernel_super_ipsec.c

```diff
@@ -174,7 +174,116 @@ static status_t del_routes_from_routing_table(private_kernel_super_ipsec_t* this
 
 static void remove_ike_sa_matching(struct pm_terminator_conn* const terminator_conn);
 static void write_terminator_tunnels_oper(struct pm_terminator_conn* const terminator_conn);
+static struct pm_terminator_conn* lookup_terminator_conn(private_kernel_super_ipsec_t* this,
+														 host_t* const host_addr);
+static host_t* _get_sp_lookup_addr(const kernel_ipsec_manage_policy_t* const data,
+								   const kernel_ipsec_policy_id_t* const id);
+
+static void write_tunnel_oper_items(private_kernel_super_ipsec_t* this,
+									kernel_ipsec_manage_policy_t* data,
+									kernel_ipsec_policy_id_t* id,
+									ike_super_peer_info_t* super_peer_info,
+									child_sa_t * child_sa,
+									bool write_uptime) {
+	if(IKE_SUPER_PEER_INFO_TYPE_TUNNEL == super_peer_info->get_type(super_peer_info)) {
+		return;
+	}
+	ike_sa_t* const ike_sa = child_sa->get_ike_sa(child_sa);
+	ike_sa_id_t * const ike_sa_id = ike_sa->get_id(ike_sa);
+	struct oper_data oper_data;
+
+	uint16_t enc_alg = ENCR_UNDEFINED, int_alg = AUTH_UNDEFINED, size;
+	uint16_t esn = NO_EXT_SEQ_NUMBERS;
+	child_sa->get_proposal(child_sa)->get_algorithm(child_sa->get_proposal(child_sa), ENCRYPTION_ALGORITHM,
+								  &enc_alg, &size);
+	child_sa->get_proposal(child_sa)->get_algorithm(child_sa->get_proposal(child_sa), INTEGRITY_ALGORITHM,
+								  &int_alg, &size);
+	child_sa->get_proposal(child_sa)->get_algorithm(child_sa->get_proposal(child_sa), EXTENDED_SEQUENCE_NUMBERS,
+								  &esn, NULL);
+
+	oper_data.encryption_algorithm = enc_alg;
+	oper_data.integrity_algorithm = int_alg;
+	oper_data.ext_seq_num = esn;
+	oper_data.dh_group = ike_sa->get_ike_cfg(ike_sa)->get_dh_group(ike_sa->get_ike_cfg(ike_sa));
+	oper_data.anti_replay_window_size = child_sa->get_config(child_sa)->get_replay_window(child_sa->get_config(child_sa));
+
+	oper_data.version = ike_sa_id->get_ike_version(ike_sa_id);
+	oper_data.inbound_ipsec_spi = child_sa->get_spi(child_sa, 1);
+	oper_data.outbound_ipsec_spi = child_sa->get_spi(child_sa, 0);
```

Comment:
`get_spi()` expects the 2nd parameter to be a bool, so pass true/false instead of 1/0.